tests/test_variable_manipulation.py

```diff
@@ -229,19 +226,52 @@ def test_find_or_check_datetime_variables(df_datetime):
         )
         == vars_convertible_to_dt + ["time_objTZ"]
     )
-    # vars cast as categorical
-    assert _find_or_check_datetime_variables(cat_date, variables="date_obj1_cat") == [
-        "date_obj1_cat"
+
+    # datetime var cast as categorical
+    df_datetime["date_obj1"] = df_datetime["date_obj1"].astype("category")
+    assert _find_or_check_datetime_variables(df_datetime, variables="date_obj1") == [
+        "date_obj1"
     ]
     assert (
-        _find_or_check_datetime_variables(
-            df_datetime[vars_convertible_to_dt].join(cat_date),
-            variables=vars_convertible_to_dt + ["date_obj1_cat"],
-        )
-        == vars_convertible_to_dt + ["date_obj1_cat"]
+        _find_or_check_datetime_variables(df_datetime, variables=vars_convertible_to_dt)
+        == vars_convertible_to_dt
     )
 
 
+@pytest.fixture()
+def cast_age_as_cat(df_datetime):
+    df_datetime["Age"] = df_datetime["Age"].astype("category")
```

Comment:
When re-defining values of a dataframe, I think a good safety measure is to copy the df first, not to override the original one. 

Not sure it is necessary here, but if we were to share these functions across scripts...

`df_datetime = df_datetime.copy()`